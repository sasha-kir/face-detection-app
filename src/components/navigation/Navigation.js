import React from 'react';

const Navigation = ({ onRouteChange }) => {
	return (
		<nav style={{display: "flex", justifyContent: "flex-end", marginBottom: "-70px"}}>
			<p className="f3 link dim white underline pa3 pointer">
				Home
			</p>
			<p id="sign-out" className="f3 link dim white underline pa3 pointer" onClick={() => onRouteChange("sign-in")}>
				Sign Out
			</p>
		</nav>
	);
}

export default Navigation;