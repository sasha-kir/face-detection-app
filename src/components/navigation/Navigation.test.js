import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

const onRouteChange = jest.fn();

describe('Navigation', () => {
  it('renders correctly', () => {
    const component = shallow(<Navigation />);
    expect(component).toMatchSnapshot();
  });
  it('performs sign out', () => {
    const component = shallow(<Navigation onRouteChange={onRouteChange} />);
    component.find('#sign-out').simulate('click');
    expect(onRouteChange).toHaveBeenCalledWith('sign-in');
  });
});