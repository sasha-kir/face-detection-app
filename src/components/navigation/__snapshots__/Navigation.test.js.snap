// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  style={
    Object {
      "display": "flex",
      "justifyContent": "flex-end",
      "marginBottom": "-70px",
    }
  }
>
  <p
    className="f3 link dim white underline pa3 pointer"
  >
    Home
  </p>
  <p
    className="f3 link dim white underline pa3 pointer"
    id="sign-out"
    onClick={[Function]}
  >
    Sign Out
  </p>
</nav>
`;
